### ISSUE TITLE
Linter is treating ClojureScript as Clojure

### ISSUE THREAD

**Issue by oconnor0** [ToxicFrog/SublimeLinter-contrib-clj-kondo#1]
I have a ClojureScript file:

```cljs
(enable-console-print!)
(js/alert "hi")
```

I saved this file as _lint-me.cljs_. SublimeText treats this file as Clojure, not as ClojureScript. With SublimeLinter-contrib-clj-kondo installed, I get an error on line 1 and a warning on line 2. When I run `clj-kondo --lint lint-me.cljs`, I get neither of those messages. If I rename the file to _lint-me.clj_ and rerun `clj-kondo` manually, I get the same error and warning I get from the linter in ST.

I tried installing EnhancedClojure to treat the _.cljs_ as ClojureScript but had no luck.

This happens in ST 3211 and 4074. I installed the linter today.

So am I doing something wrong or do I have something misconfigured?

**Comment by oconnor0** [comment]
Of course, maybe the real solution is to modify the syntax definitions to treat _.cljs_ as ClojureScript.

**Comment by ToxicFrog** [comment]
The plugin depends entirely on Sublime's understanding of what language the file is; in particular, clj-kondo itself does not see the name of the file, just the contents (this is what allows it to lint-as-you-type rather than waiting until you save), so the plugin relies on whether ST3 thinks the file is Clojure or ClojureScript to decide which language to tell the linter to lint.

So, things to check:
- ctrl-shift-p `ssyn cljs` -- see if "set syntax: clojurescript" is an option, and if so, if selecting it fixes the problem.
- install Scope Hunter through package control and use the "scope hunter: scope under cursor" command to confirm that it's `source.clojurescript` and not `source.clojure`

If you don't have "syntax: clojurescript" as an option, or if setting it doesn't change the scope to `source.clojurescript`, this is an editor config issue, and I'm not sure how to fix it -- it's worth noting that I *don't* have Enhanced Clojure installed, ST3 just automatically detects .cljs files as ClojureScript.

If your file has the `source.clojurescript` scope and it's not working, that's a bug in the plugin that I need to fix.

**Comment by oconnor0** [comment]
Yeah, I don't have a "Set Syntax: ClojureScript" option. Any chance it's being provided by another package you have installed?

**Comment by ToxicFrog** [comment]
Mystery solved -- it looks like the SublimeREPL package comes with a `clojurescript.tmlanguage` file that just includes all the rules for `source.clojure` and then overrides the scope.

So, this raises a problem -- cljs linting will only work for people with SublimeREPL installed, which is not ideal. Preferably, the solution would be for ST3 to include support for the `clojurescript` scope separate from the `clojure` scope, but I don't know how long that'll take to happen and don't want to rely on it.

I could have the plugin check the filename and treat `cljs` differently from `clj`, but (a) this is harder than using the scope, since it's not part of the linter API and (b) it doesn't work at all for buffers that don't have names, or have unusual extensions.

Possibly the solution is just to include a `clojurescript.tmLanguage` file of my own? If you're willing to help test this, could you put [this tmLanguage file](https://gist.github.com/ToxicFrog/2b870691516fd900fc58954539154be2) in `~/.config/sublime-text-3/Packages/SublimeLinter-clj-kondo`, and then see if "ClojureScript" shows up as a syntax option (either immediately or after restarting ST3)? If that works I'll just start shipping this file with the linter plugin.

**Comment by oconnor0** [comment]
Ah, wonderful, thank you.

While including `clojurescript.tmLanguage` here solves some of the problem, I'm not sure it's the "right" solution. I'm looking at the official [Clojure language package ](https://github.com/sublimehq/Packages/tree/master/Clojure) and a few other things to see if there's a more general solution to this problem that will solve more than just this 1 issue.

**Comment by ToxicFrog** [comment]
Oh, if it can be fixed upstream, that would be great -- I'm just wondering what the best workaround is for the linter plugin in the meantime.

**Comment by oconnor0** [comment]
Well, I opened an [upstream PR](https://github.com/sublimehq/Packages/pull/2355) so we'll see what happens there. In the meantime, I'm working on a standalone package that could provide Clojure and ClojureScript support.

**Comment by ToxicFrog** [comment]
In that case, I think we can safely say "this will be fixed either by your PR or by your package" and close this once one of those lands.

**Comment by oconnor0** [comment]
It does look like the `source.clojure.clojurescript` would be good to add to the list of supported scopes for ClojureScript.

**Comment by oconnor0** [comment]
I thought I had properly tested the linter last night, but it appears that having a `source.clojure` linter and a `source.clojure.clojurescript` linter means that both linters get run on ClojureScript code.
Opened https://github.com/SublimeLinter/SublimeLinter/issues/1737 to track how SublimeLinter handles this.

**Comment by oconnor0** [comment]
Never mind, scope selectors provide an easy way around that. 😀  🤦
